src/Orleans.Core/Providers/IGrainStorage.cs

```diff
@@ -31,6 +48,12 @@ public interface IGrainStorage
         /// <param name="grainState">Copy of last-known state data object for this grain.</param>
         /// <returns>Completion promise for the Delete operation on the specified grain.</returns>
         Task ClearStateAsync(string grainType, GrainReference grainReference, IGrainState grainState);
+
+        /// <summary>
+        /// Get all entries in storage
+        /// </summary>
+        /// <returns>The entries in storage</returns>
+        IAsyncEnumerable<StorageEntry> GetAll(CancellationToken cancellationToken);
```

Comment:
Should this cancellation token be marked with the `[EnumeratorCancellation]` attribute?

Reply:
I wonder if there should be an optional "fromKey" parameter so that enumeration can skip already-processed keys. Maybe it's not necessary at this stage.

Reply:
I think `ReadAllAsync` would align better with other method names

Reply:
I think we dont need it on the interface type, since it requires the specific implementation of the method:
![image](https://github.com/user-attachments/assets/23c073af-4394-4420-87ea-147552e1c691)

So we should use it on every single method implementation, but not on the interface level. Also @benjaminpetit  added it in the `Orleans.Storage.AzureBlobGrainStorage` in example